### DISCUSSION TITLE
Module not found: Can't resolve '../artifacts/contracts/Market.sol/NFTMarket.json'

### DISCUSSION THREAD

**Discussion by BarryHewitt** [vercel/next.js#28074] [Help]
Hi

I'm following along with a tutorial and I'm getting the following error 

Module not found: Can't resolve '../artifacts/contracts/Market.sol/NFTMarket.json'

The tutorial I'm following can be found here, and so can the entire code base

https://dev.to/dabit3/building-scalable-full-stack-apps-on-ethereum-with-polygon-2cfb

Once i go - npm run dev - this is my output

BarryH@EPT-LT23 MINGW64 ~/digital-marketplace (main)
$ npm run dev

> digital-marketplace@0.1.0 dev
> next dev

ready - started server on 0.0.0.0:3000, url: http://localhost:3000
info  - Using webpack 5. Reason: Enabled by default https://nextjs.org/docs/messages/webpack5
event - compiled successfully
event - build page: /
wait  - compiling...
error - ./pages/index.js:11:0
Module not found: Can't resolve '../artifacts/contracts/Market.sol/NFTMarket.json'
   9 | 
  10 | import NFT from '../artifacts/contracts/NFT.sol/NFT.json'
> 11 | import Market from '../artifacts/contracts/Market.sol/NFTMarket.json'
  12 | 
  13 | export default function Home() {
  14 |   const [nfts, setNfts] = useState([])

Import trace for requested module:

https://nextjs.org/docs/messages/module-not-found
event - build page: /next/dist/pages/_error
wait  - compiling...
error - ./pages/index.js:11:0
Module not found: Can't resolve '../artifacts/contracts/Market.sol/NFTMarket.json'
   9 | 
  10 | import NFT from '../artifacts/contracts/NFT.sol/NFT.json'
> 11 | import Market from '../artifacts/contracts/Market.sol/NFTMarket.json'
  12 | 
  13 | export default function Home() {
  14 |   const [nfts, setNfts] = useState([])

Import trace for requested module:

https://nextjs.org/docs/messages/module-not-found
wait  - compiling...
event - compiled successfully

Really don't understand why it isn't working as i've copied his code examples exactly

Any help would be greatly appreciated

**Comment by AstroCakeBSCc** [comment]
change the  name of the contract, to match the jsons i noticed this also was stuck, if you copy the one from a fork they fixed it , but NFTMarket contract and Market, when I got it to work it was MARKET

**Comment by mendsalbert** [comment]
After a couple of hours of digging I finally found the solution to this problem

1.  The 'artefacts' directory is included in the .gitignore file which makes the whole folder invisible during build
2.  open the gitignore file and exclude the artefacts directory
3. Now run 'npm run build' and everything all the error will be gone

**Reply by zialink** [reply]
Thank you very much. Your answer saved me a lot of time, I would have spent a long time on this problem.

**Comment by maaeyeng** [comment]
Module not found: Can't resolve '../artifacts/contracts/Market.sol/NFTMarket.json' 
for nextjs NFTMarketplance

**Reply by zialink** [reply]
Copy your NFTMarket.json and NFT.json files and paste them inside your nextjs directory, preferably nextjs API directory.
